datastructures/clockchain.py

```diff
@@ -32,6 +37,81 @@ def __init__(self):
         genesis_dict = self.json_tick_to_chain_tick(tick)
         self.chain.put(genesis_dict)
 
+    def fork_pool_size(self):
+        return len(self.fork_pool)
+
+    def copy_chain(self, peer_addr):
+        logger.debug("Requesting chain from peer " + str(peer_addr))
+        peer_url = self.networker.reverse_peers.get(peer_addr, None)
```

Comment:
you might want to only use 1 structure here instead of both peers and reverse_peers, future coders would get confused and forget to update one while updating the other, leading to two copies that are not representing the truth.

Reply:
I guess it depends on how big we expect the list of peers to become. Reverse lookup on a dictionary means iterating over all its values; could become costly.

Reply:
with the network topology we're going for, peers would be around 10, certainly not above 100, so a reverse lookup is probably fine. if there's two structures a future coder will run into the same confusion you had when you thought there were two kinds of ticks - a wrapper function might be the cleanest way to handle this